File: src/main/java/legacy/kms/chapter02/discount/SequenceCondition.java
package legacy.kms.chapter02.discount;

import legacy.kms.chapter02.Screening;

public class SequenceCondition implements DiscountCondition{

    private int sequence;

    public SequenceCondition(int sequence){
        this.sequence = sequence;
    }

    @Override
    public boolean isSatisfiedBy(Screening screening) {
        return screening.isSequence(sequence);
    }
}

File: src/main/kotlin/yangsooplus/ch05/discountcondition/SequenceCondition.kt
package yangsooplus.ch05.discountcondition

import yangsooplus.ch05.Screening

class SequenceCondition(
    private val sequence: Int
): DiscountCondition {

    override fun isSatisfiedBy(screening: Screening): Boolean {
        return sequence == screening.sequence
    }

}